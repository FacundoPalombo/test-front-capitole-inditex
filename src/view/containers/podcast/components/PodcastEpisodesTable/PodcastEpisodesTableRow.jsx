import React, { useMemo } from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.module.scss';

const PodcastEpisodesTableRow = ({
  trackId,
  trackName,
  releaseDate,
  trackTimeMillis,
}) => {
  const { trackMinutes, trackSeconds } = useMemo(
    () => ({
      trackMinutes: (trackTimeMillis - (trackTimeMillis % 60000)) / 60000,
      trackSeconds: ((trackTimeMillis % 60000) / 1000).toFixed(0),
    }),
    [trackTimeMillis]
  );

  const trackDateCreation = useMemo(
    () => new Date(releaseDate).toLocaleDateString(),
    [trackDateCreation]
  );

  return (
    <tr className={styles.table__row} key={trackId}>
      <th className={styles.table__row_item} align="left" headers="col-title">
        <Link to={`episode/${trackId}`} className={styles.table__row_link}>
          {trackName}
        </Link>
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-date"
      >
        {trackDateCreation}
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-duration"
      >
        {trackMinutes}
        min {trackSeconds}s
      </th>
    </tr>
  );
};

export default PodcastEpisodesTableRow;
