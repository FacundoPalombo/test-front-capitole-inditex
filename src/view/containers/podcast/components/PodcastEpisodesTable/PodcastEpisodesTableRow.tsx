import React, { useMemo } from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.module.scss';

const PodcastEpisodesTableRow = ({
  trackId,
  trackName,
  releaseDate,
  trackTimeMillis,
}) => {
  const trackMinutes = useMemo(
    () => (trackTimeMillis - (trackTimeMillis % 60000)) / 60000,
    [trackTimeMillis]
  );
  const trackSeconds = useMemo(
    () => ((trackTimeMillis % 60000) / 1000).toFixed(0),
    [trackTimeMillis]
  );
  const trackDateCreation = useMemo(
    () => new Date(releaseDate).toLocaleDateString(),
    [releaseDate]
  );

  return (
    <tr className={styles.table__row} key={trackId}>
      <th className={styles.table__row_item} align="left" headers="col-title">
        <Link to={`episode/${trackId}`} className={styles.table__row_link}>
          {trackName}
        </Link>
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-date"
      >
        {trackDateCreation}
      </th>
      <th
        className={styles.table__row_item}
        align="right"
        style={{ textAlign: 'left' }}
        headers="col-duration"
      >
        {trackMinutes}
        min {trackSeconds}s
      </th>
    </tr>
  );
};

export default PodcastEpisodesTableRow;
