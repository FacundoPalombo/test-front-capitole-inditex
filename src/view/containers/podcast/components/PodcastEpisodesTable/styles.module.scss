.table {
  padding: 16px;
  border-collapse: collapse;
  width: 100%;

  &__container {
    display: block;
    width: 100%;
    height: auto;
  }
  &__header {
    border-bottom: 2px solid rgba(170, 170, 170, 0.66);
  }
  &__title {
    padding: 16px 4px;
  }
  &__body {
  }
  &__row {
    &:nth-child(2n + 1) {
      background-color: rgba(204, 204, 204, 0.16);
    }
    border-bottom: 2px solid rgba(170, 170, 170, 0.33);
    &_link {
      text-decoration: none;
      color: var(--link);
      display: block;
      width: 100%;
      height: 100%;
      &:visited {
        color: var(--link-visited);
      }
    }
  }
  &__row_item {
    padding: 8px 4px;
  }
}
