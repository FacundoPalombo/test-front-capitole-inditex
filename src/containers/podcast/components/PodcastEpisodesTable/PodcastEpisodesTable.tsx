import { number, shape, string, arrayOf } from 'prop-types';
import React from 'react';
import styles from './styles.module.scss';
import PodcastEpisodesTableRow from './PodcastEpisodesTableRow';
import { Episode } from '@utils/types/Episodes';

type PodcastEpisodesTableProps = {
  podcasts: Episode[];
};

const PodcastEpisodesTable = ({ podcasts }: PodcastEpisodesTableProps) => {
  return (
    <div className={styles.table__container}>
      <table className={styles.table}>
        <thead className={styles.table__header}>
          <tr>
            <th className={styles.table__title} align="left" id="col-title">
              Title
            </th>
            <th
              className={styles.table__title}
              align="right"
              style={{ textAlign: 'left' }}
              id="col-date"
            >
              Date
            </th>
            <th
              className={styles.table__title}
              align="right"
              style={{ textAlign: 'left' }}
              id="col-duration"
            >
              Duration
            </th>
          </tr>
        </thead>
        <tbody className={styles.table__body}>
          {podcasts?.map((episode) => (
            <PodcastEpisodesTableRow
              key={episode.trackId!}
              trackId={episode.trackId!}
              trackName={episode.trackName!}
              releaseDate={episode.releaseDate!}
              trackTimeMillis={episode.trackTimeMillis!}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

PodcastEpisodesTable.propTypes = {
  podcasts: arrayOf(
    shape({
      trackId: number,
      trackName: string,
      releaseDate: string,
      trackTimeMillis: number,
    })
  ),
};

export default PodcastEpisodesTable;
