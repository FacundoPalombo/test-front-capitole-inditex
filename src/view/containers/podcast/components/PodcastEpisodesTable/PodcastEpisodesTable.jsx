import PropTypes from 'prop-types';
import React from 'react';
import styles from './styles.module.scss';
import PodcastEpisodesTableRow from './PodcastEpisodesTableRow';

const PodcastEpisodesTable = ({ podcasts }) => {
  return (
    <div className={styles.table__container}>
      <table className={styles.table}>
        <thead className={styles.table__header}>
          <tr>
            <th
              className={styles.table__title}
              itemScope="col"
              align="left"
              id="col-title"
            >
              Title
            </th>
            <th
              className={styles.table__title}
              align="right"
              itemScope="col"
              style={{ textAlign: 'left' }}
              id="col-date"
            >
              Date
            </th>
            <th
              className={styles.table__title}
              align="right"
              itemScope="col"
              style={{ textAlign: 'left' }}
              id="col-duration"
            >
              Duration
            </th>
          </tr>
        </thead>
        <tbody className={styles.table__body}>
          {podcasts.map((episode) => (
            <PodcastEpisodesTableRow
              key={episode.trackId}
              trackId={episode.trackId}
              trackName={episode.trackName}
              releaseDate={episode.releaseDate}
              trackTimeMillis={episode.trackTimeMillis}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

PodcastEpisodesTable.propTypes = {
  podcasts: PropTypes.shape({
    map: PropTypes.func,
  }),
};

export default PodcastEpisodesTable;
