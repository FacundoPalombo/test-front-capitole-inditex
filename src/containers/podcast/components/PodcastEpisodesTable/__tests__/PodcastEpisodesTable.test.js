import React from 'react';
import { render } from '@testing-library/react';
import PodcastEpisodesTable from '../PodcastEpisodesTable';

jest.mock('../PodcastEpisodesTableRow', () => () => (
  <div id="mockPodcastEpisodesTableRow">PodcastEpisodesTableRow</div>
));

describe('PodcastEpisodesTable unit tests', () => {
  it('should match snapshot', () => {
    const mockTable = [
      {
        key: 'keyMock',
        trackId: 'trackIdMock',
        trackName: 'trackNameMock',
        releaseDate: 'releaseDateMock',
        trackTimeMillis: 'trackTimeMillisMock',
      },
    ];
    const { asFragment } = render(
      <PodcastEpisodesTable podcasts={mockTable} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
