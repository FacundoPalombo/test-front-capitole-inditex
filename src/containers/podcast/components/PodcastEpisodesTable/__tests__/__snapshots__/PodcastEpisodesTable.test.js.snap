// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastEpisodesTable unit tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="table__container"
  >
    <div
      id="mockPodcastEpisodesTableRow"
    >
      PodcastEpisodesTableRow
    </div>
    <table
      class="table"
    >
      <thead
        class="table__header"
      >
        <tr>
          <th
            align="left"
            class="table__title"
            id="col-title"
          >
            Title
          </th>
          <th
            align="right"
            class="table__title"
            id="col-date"
            style="text-align: left;"
          >
            Date
          </th>
          <th
            align="right"
            class="table__title"
            id="col-duration"
            style="text-align: left;"
          >
            Duration
          </th>
        </tr>
      </thead>
      <tbody
        class="table__body"
      />
    </table>
  </div>
</DocumentFragment>
`;
